mesheryctl/internal/cli/root/workspaces/create.go

```diff
@@ -0,0 +1,120 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workspaces
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var CreateWorkspaceCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a new workspace",
+	Long:  `Create a new workspace by providing the name, description, and organization ID.`,
+	Example: `
+// Create a new workspace
+mesheryctl exp workspace create --orgId [orgId] --name [name] --description [description]
+
+// Documentation for workspace can be found at:
+https://docs.layer5.io/cloud/spaces/workspaces/
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		// Check prerequisites
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		if err := utils.IsServerRunning(mctlCfg.GetBaseMesheryURL()); err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return system.ErrGetCurrentContext(err)
+		}
+		if err := ctx.ValidateVersion(); err != nil {
+			return err
+		}
+		return nil
+	},
+
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Check if all three flags are set
+		orgIdFlag, _ := cmd.Flags().GetString("orgId")
+		nameFlag, _ := cmd.Flags().GetString("name")
+		descriptionFlag, _ := cmd.Flags().GetString("description")
+
+		if orgIdFlag == "" || nameFlag == "" || descriptionFlag == "" {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return utils.ErrInvalidArgument(errors.New("Please provide a --orgId, --name, and --description flag"))
+		}
+
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+
+		baseUrl := mctlCfg.GetBaseMesheryURL()
+		url := fmt.Sprintf("%s/api/workspaces", baseUrl)
+
+		nameFlag, _ := cmd.Flags().GetString("name")
+		descriptionFlag, _ := cmd.Flags().GetString("description")
+		orgIdFlag, _ := cmd.Flags().GetString("orgId")
+
+		payload := &models.WorkspacePayload{
+			Name:           nameFlag,
+			Description:    descriptionFlag,
+			OrganizationID: orgIdFlag,
+		}
+
+		payloadBytes, err := json.Marshal(payload)
+		if err != nil {
+			return err
+		}
+
+		req, err := utils.NewRequest(http.MethodPost, url, bytes.NewBuffer(payloadBytes))
+		if err != nil {
+			return err
+		}
+
+		_, err = utils.MakeRequest(req)
+
+		if strings.Contains(err.Error(), "201") {
```

Comment:
1. Flaky logic and,
2. Incorrect as well.

If there's an error, how and why is the workspace created? How the response code is `201`.
You are ignoring the response instead, use it to check the StatusCode.